### PR TITLE
Separate Eval Batch Size

### PR DESCRIPTION
**Pull request by pomonam**
This PR separates batch size for evaluation raised #161 . Specifically, each workload defines its own eval batch size. However, in case there is OOM due to the submitter having a lot of optimizer states in memory, they can optionally define `get_eval_batch_size` in the submission file that overrides the default setting.

### CODE REVIEW

File: algorithmic_efficiency/workloads/mnist/workload.py [hunk 1]
```diff
@@ -44,6 +44,10 @@ def num_validation_examples(self):
   def num_test_examples(self):
     return 10000
 
+  @property
+  def eval_batch_size(self):
+    return 1024
```
Comment (runame) [inline algorithmic_efficiency/workloads/mnist/workload.py:49]: Here we could probably fit the whole eval datasets into memory?

File: algorithmic_efficiency/workloads/imagenet_resnet/workload.py [hunk 2]
```diff
@@ -37,6 +37,10 @@ def num_validation_examples(self):
   def num_test_examples(self):
     return 10000  # ImageNet-v2
 
+  @property
+  def eval_batch_size(self):
+    return 512
```
[the remark this answers is not in the document]
Comment (pomonam) [reply algorithmic_efficiency/workloads/imagenet_resnet/workload.py]: Fixed!

File: algorithmic_efficiency/workloads/fastmri/workload.py [hunk 3]
```diff
@@ -35,6 +35,10 @@ def num_validation_examples(self):
   def num_test_examples(self):
     return 3581
 
+  @property
+  def eval_batch_size(self):
+    return 8
```
[the remark this answers is not in the document]
Comment (pomonam) [reply algorithmic_efficiency/workloads/fastmri/workload.py]: Fixed!

File: algorithmic_efficiency/workloads/cifar/workload.py [hunk 4]
```diff
@@ -32,6 +32,10 @@ def num_validation_examples(self):
   def num_test_examples(self):
     return 10000
 
+  @property
+  def eval_batch_size(self):
+    return 128
```
[the remark this answers is not in the document]
Comment (pomonam) [reply algorithmic_efficiency/workloads/cifar/workload.py]: I modified it to 1024 for now, but we can increase it if the larger is possible.

File: algorithmic_efficiency/workloads/mnist/workload.py (hunk 1 above)
Comment (pomonam) [reply algorithmic_efficiency/workloads/mnist/workload.py]: Fixed! I modified it to 10000.

**Review by znado — APPROVED** [review]

**Pull request merged**